File: A_Adrenaline_Rush.py
def main():
    import sys
    input = sys.stdin.read
    data = input().split()
    
    n = int(data[0])  # First input is `n`
    if n == 1:
        # If n = 1, no pairs can be formed
        print(0)
        return
    
    arr = list(map(int, data[1:]))  # Remaining inputs are the array elements

    if n == 2:
        # If n = 2, only one pair can exist
        print(2)
        print(arr[0], arr[1])
        print(arr[1], arr[0])
        return

    # For n > 2, calculate n + 1 pairs
    print(n + 1)

    pairs = []

    # Create pairs between other elements (excluding the first element initially)
    for i in range(1, n):
        for j in range(i + 1, n):
            pairs.append((arr[i], arr[j]))

    # Add a final pair where the first element wins indirectly
    pairs.append((arr[1], arr[0]))

    # Print the pairs
    for p in pairs:
        print(p[0], p[1])


if __name__ == "__main__":
    main()

File: A_Adrenaline_Rush.cpp
#include<bits/stdc++.h>
using namespace std;
int main(){
ios_base::sync_with_stdio(false);
cin.tie(NULL);cout.tie(NULL);
     int n;
    cin >> n;

    if (n == 1) {
       
        cout << 0 << endl;
        return 0;
    }

    vector<int> arr(n);
    for (int i = 0; i < n; i++) {
        cin >> arr[i];
    }

    if (n == 2) {
     
        cout << 2 << endl;
        cout << arr[0] << " " << arr[1] << endl;
        cout << arr[1] << " " << arr[0] << endl;
        return 0;
    }

   else if(n>2){
    cout << n + 1 << endl;

    vector<pair<int, int>> pairs;

    
    for (int i = 1; i < n; i++) {
        for (int j = i + 1; j < n; j++) {
            pairs.push_back({arr[i], arr[j]});
        }
    }


    pairs.push_back({arr[1], arr[0]});

    for (auto &p : pairs) {
        cout << p.first << " " << p.second << endl;
    }
   }

    return 0;
}